test/contracts-network/colony.js

```diff
@@ -118,6 +123,53 @@ contract("Colony", (accounts) => {
       expect(domain.skillId).to.eq.BN(rootLocalSkillId);
     });
 
+    it("should be able to make arbitrary transactions", async () => {
+      const action = await encodeTxData(token, "mint", [WAD]);
+      const balancePre = await token.balanceOf(colony.address);
+
+      await colony.makeArbitraryTransaction(token.address, 0, action);
+
+      const balancePost = await token.balanceOf(colony.address);
+      expect(balancePost.sub(balancePre)).to.eq.BN(WAD);
+    });
+
+    it("should not be able to make arbitrary transactions if not root", async () => {
+      const action = await encodeTxData(token, "mint", [WAD]);
+
+      await checkErrorRevert(colony.makeArbitraryTransaction(token.address, 0, action, { from: accounts[1] }), "ds-auth-unauthorized");
+    });
+
+    it("should not be able to make arbitrary transactions to network or token locking", async () => {
+      const tokenLockingAddress = await colonyNetwork.getTokenLocking();
+      const tokenLocking = await ITokenLocking.at(tokenLockingAddress);
+
+      const action1 = await encodeTxData(colonyNetwork, "addSkill", [0]);
+      const action2 = await encodeTxData(tokenLocking, "lockToken", [token.address]);
+
+      await checkErrorRevert(colony.makeArbitraryTransaction(colonyNetwork.address, 0, action1), "colony-cannot-target-network");
+      await checkErrorRevert(colony.makeArbitraryTransaction(tokenLocking.address, 0, action2), "colony-cannot-target-token-locking");
+    });
+
+    it("should not be able to make arbitrary transactions to transfer tokens", async () => {
+      const action1 = await encodeTxData(token, "approve", [USER0, WAD]);
+      const action2 = await encodeTxData(token, "transfer", [USER0, WAD]);
+
+      await checkErrorRevert(colony.makeArbitraryTransaction(token.address, 0, action1), "colony-cannot-call-erc20-approve");
+      await checkErrorRevert(colony.makeArbitraryTransaction(token.address, 0, action2), "colony-cannot-call-erc20-transfer");
+    });
+
+    it("should not be able to make arbitrary transactions to extensions", async () => {
```

Comment:
There should be a test demonstrating that it can make arbitrary transactions to extensions installed in other colonies, or (equivalently) something that has the `identifier()` function but isn't an extension.